### DISCUSSION TITLE
I've hit a wall

### DISCUSSION THREAD

**Discussion by darkstarfish1337** [mickem/nscp#792] [Q&A]
Hello,

I'm still learning, but it seems that I missing something here. I hope you can confirm this for me.  I'm testing configs with an icinga2 Server and got the monitoring to work with Linux (at least via checks that are interesting for me). 

I'm trying to use nsclient++ similarly to the Linux setup, and have the local nsclien.ini override the defaults set on the server. I can get the client to run, and report ping, load and frer drive space no problem. I would however like to change the warning and critical levels on the windows VM. For the life of me, I can't get change the levels. I've been searching for a few days and there are only fancy alias examples which I honestly don't understand. 

From the icinga2 server I can pass the following command  with success (default is -w 20% -c 10% ): 

`/usr/lib/nagios/plugins/check_nrpe -H 10.100.100.72 -c check_drivesize -a warning=free<2% critical=free<1% drive=c:`

How do I pass this command in the nsclient.ini config? 

Any help/nudge/beating would be greatly appreciated!

**Comment by mickem** [comment]
Sorry for the very late reply. The thing you are looking for is an **alias**: a named command defined in `nsclient.ini` that runs a built-in check with fixed arguments. That is what the "fancy alias examples" are, and the minimal form is one line:

```ini
[/settings/external scripts/alias]
alias_disk_c = check_drivesize drive=c: "warning=free < 2%" "critical=free < 1%"
```

Then call the alias instead of `check_drivesize`, with no `-a` arguments at all:

```
/usr/lib/nagios/plugins/check_nrpe -H 10.100.100.72 -c alias_disk_c
```

Notes:

- Restart the service (or run `nscp service --restart`) after editing the file, and make sure the `CheckExternalScripts` module is enabled under `[/modules]` (the default install enables it; aliases live in that module).
- The quoting is only needed because the arguments contain spaces; `warning=free<2%` without spaces works unquoted, exactly as on your `check_nrpe` line.
- Aliases are the recommended way to do this because NRPE's `allow arguments` can then stay off on the agent (`[/settings/NRPE/server] allow arguments = false`), so nobody on the network can send the agent other thresholds. If you keep passing `-a` from Icinga instead, `allow arguments = true` (and `allow nasty characters = true` for `<`, `>` and friends) must be set on the agent.
- You can define as many aliases as you like (`alias_disk_d`, `alias_cpu`, ...); each is just a name in front of a full command line.
